controllers/UsersController.js

```diff
@@ -1,92 +1,92 @@
-
-// const logger = message => (debug('dev')(message));
+const users_model = new Model('users');
 
 const UsersController = {
 
-    dashboard: (req, res) => {
-        res.render( 'dashboard', { title: 'Dashboard' });
+    get_user: (req, res) => {
+        const { id } = req.params;
+        (async () => {
+            try {
+                const { rows } = await users_model.select(
+                    `id, email, password, firstname,
+                    lastname, phone, status, address`,
+                    `id=${id}`
+                );
+                if (rows.length === 0) {
+                    // user was not found
+                    return res.status(404).json({
+                        error: `User with id ${id} not found`
+                    });
+                }
+                return res.status(200).json({ data: rows[0] });
+            }
+            catch (e) {
+                return InternalServerError(req, res, e);
+            }
+        })();
     },
-    
+
     verify_user: (req, res) => {
-        const { email } = req.params;
-        const user = users.find(user => (user.email === email));
-        
-        if (user === undefined) {
-            res.status(404).json({ error: 'User not found' });
-        }
-        else {
-            user.status = 'verified';
-            res.status(200).json({ data: user });
-        }
+        const { id } = req.params;
+        (async () => {
+            try {
+                await users_model.update(
+                    'status=\'verified\'',
+                    `id=${id}`
+                );
+                UsersController.get_user(req, res);
+            }
+            catch (e) { return InternalServerError(req, res, e); }
+        })();
     },
-    
+
     get_users: (req, res) => {
         const { status } = req.query;
-        
-        if (status) {
-            const data = users.filter(user => (user.status === status));
-            res.status(200).json({ data });
-        }
-        else {
-            const data = [];
-            users.map(user => data.push(user));
-            res.status(200).json({ data });
-        }
+        (async () => {
+            try {
+                let data;
+                if (status) {
+                    data = await users_model.select(
+                        `id, email, password, firstname,
+                    lastname, phone, status, address`,
+                        `status='${status}'`
+                    );
+                }
+                else {
+                    data = await users_model.select(
+                        `id, email, password, firstname,
+                        lastname, phone, status, address`,
+                    );
+                }
+                return res.status(200).json({ data: data.rows });
+            }
+            catch (e) { return InternalServerError(req, res, e); }
+        })();
     },
 
-    get_user: (req, res) => {
+    update_user: (req, res) => {
```

Comment:
Make this function async so you can do await within its body

Reply:
Done